File: py/Garage.py
from Vehicle import Vehicle

class Garage:
    def __init__(self, nama_garasi, luas_garasi, daftar_kendaraan):
        self.nama_garasi = nama_garasi
        self.luas_garasi = luas_garasi
        self.daftar_kendaraan = daftar_kendaraan

    def __str__(self):
        return f"""
Nama Garasi: {self.nama_garasi}
Luas Garasi: {self.luas_garasi}

Daftar Kendaraan:
{''.join([str(kendaraan) for kendaraan in self.daftar_kendaraan])}
"""

File: cpp/garage.cpp
#pragma once
#include <iostream>
#include <vector>
#include <iomanip>
#include "vehicle.cpp"

using namespace std;

class Garage
{
    public:
        string nama_garasi;
        int luas_garasi;
        vector<Vehicle*> daftar_kendaraan;

        Garage(string nama_garasi, int luas_garasi, vector<Vehicle*> daftar_kendaraan) : 
            nama_garasi(nama_garasi), luas_garasi(luas_garasi), daftar_kendaraan(daftar_kendaraan)
        {
        }

        string toString()
        {
            string result = "Nama Garasi: " + nama_garasi + "\nLuas Garasi: " + to_string(luas_garasi) + "\n\nDaftar Kendaraan:\n";
            for (Vehicle* kendaraan : daftar_kendaraan)
            {
                result += kendaraan->toString();
            }
            return result;
        }
};